Adds getting started in Readme.MD

mustafaiman opened the pull request.

furkansenharputlu commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+
+These configuration elements are enough for most connection scenarios. Now we will move onto configuration of the Node.js client.
+
+## Hazelcast Client Configuration
```
This part is similar with configuration options in the PR #342 .

mustafaiman replied:
This part looks like repeated in more than two places. I think redundancy in a guide is not a bad thing though. I say let us keep redundant parts for now. If we find them distracting or completely useless, we can remove them at the end.

Serdaro commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+two formats are different, you will realize that the names of the configuration parameters are the same for both client and the member.
+It is done this way to make it easier to transfer Hazelcast skills to multiple platforms easily.
+
+Once you embedded hazelcast-client to your Node.js project. You may follow any of programmatic or declarative configuration approaches.
```
Once you embedded hazelcast-client to your Node.js project, you may follow...

Serdaro commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+---
+
+If you run Hazelcast IMDG members in a different server than the client, you most probably have configured the members' ports and cluster
+names as explained in the previous section. If you did, then you need to make certain changes to network settings of your client.
```
...to the network settings of...

Serdaro commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
-Hazelcast Node.js Client connects to a Hazelcast IMDG cluster. See [https://hazelcast.org/download/](https://hazelcast.org/download/).
+## Basic Usage
+Now that we have a working cluster and we know how to configure both our cluster and client, we can run a simple program to use a
+distributed map in Node.js
```
...in Node.js client.

mustafaiman commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+side to client side in order to connect to the cluster. This section describes the most common configuration elements to get you started in no time.
+It discusses some member side configuration options to ease understanding Hazelcast's ecosystem. Then, client side configuration options
+regarding cluster connection are discussed. Configuration material regarding data structures are discussed in the following sections.
+You can refer to [IMDG Documentation](https://hazelcast.org/documentation/) and [CONFIG.md](CONFIG.MD) for more information.
```
I do not think `CONFIG.md`s whole content is going to be moved here. `CONFIG.md` discusses all details of configuration, not only common cases.

mustafaiman commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+
+## IMDG Configuration
+There are two ways to configure Hazelcast IMDG. One is to use a `hazelcast.xml` file and the other is to programmatically configure the
+instance before starting it from Java code. Since we use standalone servers, we will use `hazelcast.xml` to configure our cluster members.
```
I am adding a new parahgraph just above this one about why we may want to configure standalone servers

mustafaiman commented on README.md:
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+instance before starting it from Java code. Since we use standalone servers, we will use `hazelcast.xml` to configure our cluster members.
+
+When you download and unzip hazelcast-<version>.zip, you see the `hazelcast.xml` in /bin folder. When a Hazelcast member starts, it looks for
+`hazelcast.xml` file to load configuration from. In `hazelcast.xml` file, you will see the default configuration. We will go over some important
```
absolutely good suggestion

furkansenharputlu commented on README.md:
```diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
+
+### Group Settings
+
+- Programmatic
```
Using this style is better to be compatible with other PRs.
**Programmatic:**
```javascript
let cfg = new Config.ClientConfig();
cfg.group.name = // group name of you cluster
```

**Declarative:**
```json
{
    "group": {
        "name": "group name of you cluster"
    }
}
```

furkansenharputlu commented on README.md:
````diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
+### Group Settings
+
+- Programmatic
+```
````
`javascript`

mustafaiman commented on README.md:
```diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
+
+#### Running Standalone Jars
+Go to https://hazelcast.org/download/ and download `.zip` or `.tar` distribution of Hazelcast IMDG. Decompress the contents into any directory that you
+want to run IMDG members from. Change into the directory that you decompressed the Hazelcast content. Go into `bin` directory. Use either
```
actually `/bin` is not correct. It links to a bin folder in root whereas I was referring to a bin directory in current working directory. The other one is wrong, I'll change that.

mustafaiman commented:
@furkansenharputlu I think I have addressed all the comments now. Can you take another look?
@Serdaro @furkansenharputlu I removed the reference to `CONFIG.md` and replaced them with references to `Configuration Overview` chapter. I guess it is better to incorporate all the information in `CONFIG.md` to here and get rid of it as you previously suggested.

mustafaiman commented:
@furkansenharputlu do you have further comments?

furkansenharputlu commented on README.md:
```diff
@@ -65,51 +65,351 @@ Hazelcast Node.js client supports the following data structures and features:
+Once you embedded `hazelcast-client` to your Node.js project, you may follow any of programmatic or declarative configuration approaches.
+We will provide both ways for each configuration option in this section. Pick one way and stick to it.
+
+---
```
??
